### DISCUSSION TITLE
Why isn't the `regex_replace` filter already built-in yet?

### DISCUSSION THREAD

**Discussion by brlin-tw** [pallets/jinja#1947] [Q&A]
The functionality to search and replace text using regular expression patterns seems to be very useful in a template.  While [hooking the functionality using custom filters](https://jinja.palletsprojects.com/en/3.1.x/api/#custom-filters) [definitely works](https://stackoverflow.com/questions/12791216/how-do-i-use-regular-expressions-in-jinja2) I wonder why it wasn't implemented as a built-in filter in the first place?

Thanks in advance!

**Comment by ThiefMaster** [comment]
Is it really that common? I have a somewhat large webapp (over 500 template files) and never had the need for doing a regex replacement in a template....

**Reply by brlin-tw** [reply]
Some reasoning I've gathered so far:

* It is implemented in [Ansible](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_filters.html#searching-strings-with-regular-expressions), [Salt](https://github.com/saltstack/salt/issues/48952), [Pixip Infinity](https://docs.pexip.com/admin/jinja_templates.htm#custom_filters), and other downstream products that make use of Jinja2.
* There are [more than 200 results of Stack Overflow related questions](https://stackoverflow.com/search?page=4&tab=Relevance&pagesize=50&q=%5bjinja2%5d%20regex&searchOn=3) using `[jinja2] regex` as the search keyword.

This feature is required for high-precision plaintext data manipulation and is generic enough that it should be implemented upstream as a built-in filter to benefit all users, including some (like me) that can't figure out how to make use of the custom filter mechanism due to unfamiliar with Python programming in general.

**Reply by davidism** [reply]
There's not really a reason to. Publish a package that provides the filter if you think it is generally reusable.

**Reply by brlin-tw** [reply]
> Publish a package that provides the filter if you think it is generally reusable.

I believe that this is not a practical option for the endusers in general.

**Reply by davidism** [reply]
If you don't want to take advantage of packaging and the community ecosystem, that's up to you. Perhaps someone else will publish it instead.
